packages/client/hmi-client/src/services/workflow.ts

```diff
@@ -0,0 +1,139 @@
+import { v4 as uuidv4 } from 'uuid';
+import _ from 'lodash';
+import {
+	Workflow,
+	Operation,
+	WorkflowNode,
+	WorkflowStatus,
+	Position,
+	WorkflowEdge
+} from '@/types/workflow';
+
+/**
+ * Captures common actions performed on workflow nodes/edges. The functions here are
+ * not optimized, on the account that we don't expect most workflow graphs to
+ * exceed say ... 10-12 nodes with 30-40 edges.
+ *
+ * TODO:
+ * - Should we update workflow node status on modification???
+ */
+
+export const create = () => {
+	const workflow: Workflow = {
+		id: uuidv4(),
+		name: 'test',
+		description: '',
+
+		transform: { x: 0, y: 0, k: 1 },
+		nodes: [],
+		edges: []
+	};
+	return workflow;
+};
+
+export const addNode = (wf: Workflow, op: Operation, pos: Position) => {
+	const node: WorkflowNode = {
+		id: uuidv4(),
+		workflowId: wf.id,
+		operationType: op.name,
+		x: pos.x,
+		y: pos.y,
+
+		inputs: op.inputs.map((port) => ({
+			id: uuidv4(),
+			type: port.type,
+			label: port.label,
+			value: null
+		})),
+		outputs: op.outputs.map((port) => ({
+			id: uuidv4(),
+			type: port.type,
+			label: port.label,
+			value: null
+		})),
+		statusCode: WorkflowStatus.INVALID,
+
+		// Not currently in use. May 2023
+		width: 100,
+		height: 100
+	};
+
+	wf.nodes.push(node);
+};
+
+export const addEdge = (
+	wf: Workflow,
+	sourceId: string,
+	sourceOutputPortId: string,
+	targetId: string,
+	targetInputPortId: string,
+	points: Position[]
+) => {
+	const sourceNode = wf.nodes.find((d) => d.id === sourceId);
```

Comment:
Similar question, should edge and node have IDs of strings going fwd?
Again atm I dont really care